src/trans.c

```diff
@@ -1500,16 +1500,24 @@ Obj FuncON_KERNEL_ANTI_ACTION (Obj self, Obj ker, Obj f, Obj n) {
   UInt    deg, i, j, rank, len;
   Obj     out;
 
-  if (INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0) {
+  assert(IS_LIST(ker));
+  assert(IS_INTOBJ(n));
```

Comment:
We could do this, but this is a performance critical function for the Semigroups package, so I'm not sure it's a good idea. An alternative would be to remove it from the GAP library altogether, and put it into the Semigroups package kernel module. There are a number of functions in `trans.c` that are really only intended to be used by the Semigroups package, from the time when it did not have a kernel module. Now that Semigroups does have a kernel module again it might be a good idea to remove all of these functions to the Semigroups package. I don't really have any strong feelings about that either way.